### PR TITLE
Implement Support Screen & Markdown Support

### PR DESCRIPTION
**Pull request by jawrainey**
- [x] Assigned a Reviewer to the **PR**.
- [x] Assigned a Project to the **PR**.

## What Changed and Why?

  1. Added Support for importing markdown files
  2. Implemented store state for language selector
  3. Added default state to the store for devices
  4. Designed Device component and used on the support page
  5. Added structure and use of importing and rendering the content of the markdown files inside SupportDoc.

*Note:* the content of the markdown files are currently rendered and not transformed as that's outside the scope of this PR, i.e. the raw text is shown and not marked-up HTML or such. 

## How to Review Changes?

Verify that the support screen mirrors the prior [adobe design](https://xd.adobe.com/view/d8cb1f48-e69d-4acb-b97d-9ec34c15f0b7-8d98/screen/cffc1d75-1100-4262-9da7-a9d2b552d308) and that the loading of markdown files is correct. To do that:

  1. Open the app and go to settings (Contributions->Cog) and change the language to whatever you want.
  1. Tap the Support tab button, then choose one of the devices. This will take you to a new screen and the raw content of the markdown file inside. Depending on the langage and device chosen, the content should be "DeviceName LangCode"
 1. Tap back to the "Contributions" tab and the Settings screen will still be open. Change your language and tab back to "Support" and you'll see that the other language file has been loaded

### CODE REVIEW

File: android/app/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
         <!-- Customize your theme here. -->
         <item name="android:textColor">#000000</item>
+        <item name="android:colorControlNormal">#FFFFFF</item>
```
Comment (jawrainey) [inline android/app/src/main/res/values/styles.xml:7]: Note: this changes the color of the [chevron-down](https://fontawesome.com/icons/chevron-down) icon in the LanguageSelection container from the default grey-ish to white to have a consistent UI. Changing this color is not possible via react-native, hence this change.

File: src/containers/support/SupportDoc.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import {mapDispatchToProps} from '../../ducks/actions';
+
+import AllContent from './docs';
+
+const SupportDoc: () => React$Node = props => {
+  const [content, setContent] = useState('');
+
+  useEffect(() => {
+    let device = props.route.params.device;
+    setContent(AllContent[device.id][props.userLang]);
```
Comment (farzanehfar) [inline src/containers/support/SupportDoc.js:16]: Can this one to be a const?

File: babel.config.js [hunk 3]
```diff
@@ -1,3 +1,11 @@
 module.exports = {
   presets: ['module:metro-react-native-babel-preset'],
+  plugins: [
+    [
+      'inline-import',
+      {
+        extensions: ['.md'],
+      },
+    ],
+  ],
```
Comment (farzanehfar) [inline babel.config.js:10]: Couldn't we have this as a .babelrc/ .bable?

File: src/stacks/support.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import {createStackNavigator} from '@react-navigation/stack';
 import {Stack} from '../styles';
 import React from 'react';
-import Support from '../containers/Support';
+import Support from '../containers/support/Support';
```
Comment (farzanehfar) [inline src/stacks/support.js:4]: I know there are many ways, but nested dir, was never my favourite. is up to u keep it or transfer it in independent dir.

Comment (jawrainey) [reply src/stacks/support.js:4]: What's the alternative practices to resolve this? The devices/support pages will have a lot of sub containers so if there are no folder structure it'll become a mess?

File: src/containers/LanguageSelection.js [hunk 5]
```diff
@@ -32,15 +32,19 @@ const LanguageSelection: () => React$Node = props => {
...
-    <RadioButtons options={languages} active={lang} onPress={setLanguage} />
+    <RadioButtons options={languages} active={userLang} onPress={setLanguage} />
   );
 
-  const activeLanguage = languages.filter(l => l.code === lang)[0].name;
+  useEffect(() => {
+    setUserLang(props.userLang);
+  }, [props.userLang]);
+
+  const activeLanguage = languages.filter(l => l.code === userLang)[0].name;
```
Comment (drewmiley) [inline src/containers/LanguageSelection.js:47]: Could use find?

Comment (drewmiley) [reply src/containers/LanguageSelection.js:47]: Which is basically filter but return the first one

File: src/containers/support/SupportDoc.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import {connect} from 'react-redux';
+import {mapDispatchToProps} from '../../ducks/actions';
+
+import AllContent from './docs';
+
+const SupportDoc: () => React$Node = props => {
+  const [content, setContent] = useState('');
+
+  useEffect(() => {
+    let device = props.route.params.device;
```
Comment (drewmiley) [inline src/containers/support/SupportDoc.js:15]: Does this need to be a let?

File: src/containers/support/docs/index.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
+import AxivityEN from './i18n/en/axivity.md';
```
Comment (drewmiley) [inline src/containers/support/docs/index.js:1]: Hmm... might this be better in a docs folder outside of the containers directory? At the same time, I see why you've done it and I guess you'd probably already considered that.

I think the md files (placeholder though they are) belong in a separate non-nested folder, but maybe this index.js could stop in this directory? Let me know what you think

File: src/containers/LanguageSelection.js [hunk 8]
```diff
@@ -32,15 +32,19 @@ const LanguageSelection: () => React$Node = props => {
...
 
   const setLanguage = language => {
-    setLang(language.code);
+    props.setUserLang(language.code);
     showhideModal();
   };
 
   const languageButtons = (
-    <RadioButtons options={languages} active={lang} onPress={setLanguage} />
+    <RadioButtons options={languages} active={userLang} onPress={setLanguage} />
```
Comment (drewmiley) [inline src/containers/LanguageSelection.js:40]: 'languages' could be set in the initial state file and passed in here

**Review by drewmiley — CHANGES_REQUESTED** [review]
Just a few comments/considerations - only very minor things.

File: babel.config.js (hunk 3 above)
Comment (jawrainey) [reply babel.config.js:10]: The reason I didn't was to reduce adding another file for simplicity -- happy to add a `.babel.rc` 👍🏼

**Review by drewmiley — COMMENTED** [review]
From my perspective, I am happy with this. 👍

**Review by farzanehfar — APPROVED** [review]

**Review by drewmiley — APPROVED** [review]

**Pull request merged**